lisa/runners/lisa_runner.py

```diff
@@ -194,13 +194,28 @@ def _prepare_environments(
                 matched_result = self._pick_one_result_on_environment(
                     environment=candidate_environment, results=test_results
                 )
-                if matched_result:
-                    self._attach_failed_environment_to_result(
-                        platform=platform,
-                        environment=candidate_environment,
-                        result=matched_result,
-                        exception=identifier,
+                if not matched_result:
+                    self._log.info(
+                        "no requirement meets capability of the failed environment, "
+                        "it may cause by no feature defined on customized environment. "
+                        "attach one of test case to this deployment failure."
```

Comment:
Let me try and understand what is going on here, for this case:
- *No* tests that would go into said environment fit into its provided node capabilities
- One of the causes may be that it's a custom, user-provided environment
- The test suite picks *one* test case (actually its result object) to display a failure, only

Is that correct?

Reply:
If so, can we please reword it, for better clarity?
f"No tests targeting environment {env} fit into the capabilities provided by it."
"(Is it a custom, user-provided environment? Please go check its capabilities, if so). "
"Displaying first test failure."

Reply:
Very close understanding. The picked is a test result, which doesn't run. After attached the failed deployment, the test result is from NotRun to Failed.

Good suggestion on rewording. One cent is, since the information is in log, not interactable. I don't prefer to have questions style.

Reply:
The `feature` field is special in requirement and capability. It's filled in preparing phase by platform in Azure, and it doesn't need to be specified in a customized environment in runbook. So, when platform preparing fails, the capability may not be filled yet, and it causes no case can match the environment if a case needs some feature. But if the environment is prepared well, it can be matched.